src/mx/Monitor.java

```diff
@@ -0,0 +1,40 @@
+package mx;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+
+public class Monitor {
+    private ArrayList<Domain> domainList;
+    private long delayInMs;
+
+    public Monitor(ArrayList<Domain> domainList, long delayInMs) {
+        this.domainList = domainList;
+        this.delayInMs = delayInMs;
+    }
+
+    public ArrayList<Domain> getDomainListWithStatuses() throws HostException
+    {
+        try{
+            checkDomainsStatuses();
+        } catch (InterruptedException | IOException e){
+            throw new HostException("Wystąpił błąd z pobraniem statusu dla domen");
+        }
+        return domainList;
+    }
+
+    private void checkDomainsStatuses() throws InterruptedException, IOException {
+        for (Domain domain : domainList) {
+            checkDomainStatus(domain);
+            Thread.sleep(delayInMs);
+        }
+    }
+
+    private void checkDomainStatus(Domain domain) throws IOException {
```

Comment:
This method name suggests that it only checks something, but it actually modifies it's parameters - very bad idea. Make it accept some immutable parameter and return an object.
For example, the parameter could be `String` or `URL`, and return type `DomainStatus`, as discussed before